src/Data/Fin/Mod.agda

```diff
@@ -0,0 +1,107 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- ℤ module n
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+module Data.Fin.Mod where
+
+open import Function using (_$_; _∘_)
+open import Data.Bool using (true; false)
+open import Data.Nat as ℕ using (ℕ; s≤s)
+open import Data.Nat.Properties as ℕ
+  using (m≤n⇒m≤1+n; 1+n≰n; module ≤-Reasoning)
+open import Data.Fin as F hiding (suc; pred; _+_; _-_)
+open import Data.Fin.Properties
+open import Relation.Nullary using (Dec; yes; no; contradiction)
+open import Relation.Binary.PropositionalEquality
+  using (_≡_; _≢_; refl; sym; trans; cong; module ≡-Reasoning)
+import Algebra.Definitions as ADef
+
+private variable
+  m n : ℕ
+
+open module AD {n} = ADef {A = Fin n} _≡_
+
+private
+  hs : ∀ {i : Fin (ℕ.suc n)} → i ≢ fromℕ n → Fin (ℕ.suc n)
+  hs {i = i} i≢n = lower₁ (F.suc i) help
+    where
+    help : _
+    help = i≢n ∘ sym ∘ toℕ-injective ∘ trans (toℕ-fromℕ _) ∘ cong ℕ.pred
+
+suc : Fin n → Fin n
+suc {ℕ.suc n} i with i ≟ fromℕ n
+... | yes  _ = zero
+... | no i≢n = hs i≢n
+
+pred : Fin n → Fin n
+pred zero = fromℕ _
+pred (F.suc i) = inject₁ i
+
+_ℕ+_ : ℕ → Fin n → Fin n
+ℕ.zero ℕ+ i = i
+ℕ.suc n ℕ+ i = suc (n ℕ+ i)
+
+_+_ : Fin m → Fin n → Fin n
+i + j = toℕ i ℕ+ j
+
+_+‵_ : Fin n → Fin n → Fin n
```

Comment:
Is this ever necessary over `_+_`? A lot of the time Agda should be able to unify the two dimensions.

Reply:
It could be necessary if someone wants a more restrictive version.
This is one example:
https://github.com/agda/agda-stdlib/blob/78e11bdee2eaaf7652c7be32d890821aeebe8781/src/Function/Base.agda#L134